test/new-e2e/tests/discovery/linux_test.go

```diff
@@ -344,20 +379,31 @@ func (s *linuxTestSuite) testProcessCheckWithServiceDiscovery(agentConfigStr str
 }
 
 type checkStatus struct {
-	CheckID           string `json:"CheckID"`
-	CheckName         string `json:"CheckName"`
-	CheckConfigSource string `json:"CheckConfigSource"`
-	ExecutionTimes    []int  `json:"ExecutionTimes"`
+	CheckID           string   `json:"CheckID"`
+	CheckName         string   `json:"CheckName"`
+	CheckConfigSource string   `json:"CheckConfigSource"`
+	ExecutionTimes    []int    `json:"ExecutionTimes"`
+	LastWarnings      []string `json:"LastWarnings"`
 }
 
 type runnerStats struct {
-	Checks map[string]checkStatus `json:"Checks"`
+	Checks map[string]map[string]checkStatus `json:"Checks"`
 }
```

Comment:
Small suggestion that I think might help readers to parse the type. It uses type aliases so this should not add any verbose casts to the rest of the code.

```suggestion
type checkName = string
type checkGroupName = string
type runnerStats struct {
	Checks map[checkGroupName]map[checkName]checkStatus `json:"Checks"`
}
```